webui/react/src/components/Navigation.tsx

```diff
@@ -32,18 +32,14 @@ const Navigation: React.FC<Props> = ({ children }) => {
   usePolling(fetchAgents);
   usePolling(fetchPinnedWorkspaces);
   usePolling(fetchUserSettings, { interval: 60000 });
+  usePolling(fetchKnownRoles);
```

Comment:
How often to fetch the list of available roles?  I'd recommend a longer interval (like the previous line).

I started thinking about this in #5079 - most users do not use the roles data -- only admins visiting RBAC UI. But usually we have `usePolling` here with no restriction.  I don't know if there is a way we can poll only for certain users?

Reply:
You are right, list of available roles do not need to be fetched frequently, it is the user's roles/assignments need `usePolling`

Reply:
I added fetch for current user's role in usePolling, also only call API when rbac is enabled